### ISSUE TITLE
Palette Issues

### ISSUE THREAD

**Issue by XaserAcheron** [XaserAcheron/argent#49]
Need to include some versions of paletted graphics to avoid stuff like [this](https://cdn.discordapp.com/attachments/159107819998543873/225204591820668929/Screenshot_Doom_20160913_064145.png) in Linux.

I'm a bit surprised the crosshairs are misbehaving since those are treated specially anyway, but I may have forgotten to greyscale the graphics. Will check that first.

**Comment by XaserAcheron** [comment]
@Pink-Silver, are the HAR/Gauss crosshairs any better in Linux-land? I pushed a thing but I'm not quite sure if it's what we needed.

Reminder to self to test with `vid_forceddraw` later when I get home-ish. I forgot to because I'm a derpface.

**Comment by Pink-Silver** [comment]
@XaserAcheron Negative, it's still rendering the translucency as solid, so it's still going through the conversion routine.

Not a surprise, [Greyscale isn't an indexed color type](http://www.libpng.org/pub/png/book/chapter08.html#png.ch08.div.5.3) in the PNG specification, so it may be necessary to explicitly create a greyscale palette and apply that instead.

**Comment by XaserAcheron** [comment]
Hmm, I was afraid of that. The trick, then, is figuring out exactly how to convert a PNG into an actual proper greyscaled rather than one that happens to have a grey palette (the latter is what the images probably are now). Or otherwise figure out what image type is that works. D:

It's weird though because the D3D renderer is totally fine with it. I wonder if it's a legit ZDoom bug?

**Comment by Pink-Silver** [comment]
It's the other way around, dudebro. ZDoom wants a greyscale _indexed palette_, because it uses the index as an alpha value, but your images are _actual_ greyscale, which means the palette renderer needs to apply a palette to it but winds up mistaking 0 as fully opaque because it swaps it with 247 due to the way the rendering works, as it claims a duplicate palette slot to represent transparency.

For reference, the D3D method uses a shader and does the smart thing by taking the red channel and _that_ as the alpha value, which is the sensible way of doing it, because it removes palette indexes from play all together, but it likely wasn't done that way because the DOOM palette has all of 80 greys in it, not really as much as a full greyscale palette could represent.

**Comment by XaserAcheron** [comment]
Well, that's wacky as all hell. How the fook is someone supposed to figure that out? D:

It's probably worth documenting some of this [on the wiki](https://zdoom.org/wiki/Crosshair), since it mentions nothing of the sort (and, in fact, seems to suggest proper-greyscale if the language is taken literally).

Will try another fix here in a second.

**Comment by XaserAcheron** [comment]
Well, that didn't work. Indexed-palette PNG shows the same issues as greyscale PNG. Tried Truecolor too just to be goofy; still no dice.

**Comment by XaserAcheron** [comment]
Welp. Between this and the fact that [PALVERS doesn't work on HUD graphics (WTF)](http://forum.zdoom.org/viewtopic.php?f=15&t=33297&p=951331#p951331), I'm going to "officially" put this on hold. Not closing it outright, but removing it from the v0.3.0 milestone.

I reeeeeally hate to tell non-Windows folk "use GZDoom" (or QZDoom now? :P), but it's still all technically in alpha, so... D:

I'll fix the crosshair issue once a sane solution comes forth.

**Comment by Pink-Silver** [comment]
Yeah I don't really have the tools to properly fix this right now either, so I apologize. It requires an editor with strict control over PNG chunks and palette-editing capabilities.

**Comment by XaserAcheron** [comment]
No worries -- thanks for the help there!

I'll open a question on the ZDoom Editing forum for now. Better than doing nothing at least.

[EDIT] For future reference, [here's the thread](https://forum.zdoom.org/viewtopic.php?f=3&t=54069). I'm fairly convinced at this point that PNG support for crosshairs is broken, so I'll probably just end up converting the crosshairs to IMGZ.
